src/app/ui/components/containers/dialog/dialog.tsx

```diff
@@ -0,0 +1,123 @@
+import { ReactNode, memo } from 'react';
+
+import * as RadixDialog from '@radix-ui/react-dialog';
+import { css } from 'leather-styles/css';
+import { Box } from 'leather-styles/jsx';
+
+import { Header } from '@app/ui/components/containers/headers/header';
+
+export interface DialogProps {
+  children?: ReactNode;
+  footer?: ReactNode;
+  isShowing: boolean;
+  isWaitingOnPerformedAction?: boolean;
+  onGoBack?(): void;
+  onClose(): void;
+  canClose?: boolean;
+  pauseOnClickOutside?: boolean; // FIXME - seem to have lost what this does in useDrawer() I think it blocked close on click outside?
```

Comment:
That was my understanding of its use. I think it can be replaced if we aren't using the drawer any longer, but does `canClose` prevent the user from being able to use the `X` button as well? I think the other was just for when a user clicks off the open drawer, but they could still close it with the close button?

Reply:
Thanks! I can make sure that if the `X` is available they can still click that 👍

Reply:
Always found this confusing as it's a very ledger specific concern added to a generic dialog. Remove if you can.